pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -70,15 +69,14 @@ func (a ByLabelName) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 // tsMap will be unmodified if either labels or sample is nil, but can still be modified if the exemplar is nil.
 func addSample(tsMap map[string]*prompb.TimeSeries, sample *prompb.Sample, labels []prompb.Label,
 	datatype string) string {
-
 	if sample == nil || labels == nil || tsMap == nil {
+		// This shouldn't happen
 		return ""
 	}
 
-	sig := timeSeriesSignature(datatype, &labels)
-	ts, ok := tsMap[sig]
-
-	if ok {
+	sig := timeSeriesSignature(datatype, labels)
+	ts := tsMap[sig]
+	if ts != nil {
```

Comment:
Is there a performance improvement doing it this way? I have always relied on the `ok` returned by the map and was under the impression it was a common practice.

Reply:
This change is not for performance, but for robustness. Just testing `ok` doesn't also handle the case where `ts == nil`. To achieve the same with `ok`, you'd have to do the following:

```
ts, ok := tsMap[sig]

if ok && ts != nil {
```

The question is what's the point of `ok`, since checking whether `ts` is nil covers both cases you have to look out for:
* `sig` is not in `tsMap` => ts == nil
* `sig` is in `tsMap`, but with a nil value => ts == nil

I think `ok` is only useful when the type's zero value isn't suitable to check for (i.e., if the key is missing from the map, you get a zero value). Otherwise, it's just superfluous.

Make sense?